File: ScoutTrack/ScoutTrack.Services/Database/Entities/Review.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ScoutTrack.Services.Database.Entities
{
    public class Review
    {
        [Key]
        public int Id { get; set; }

        public string Content { get; set; } = string.Empty;
        public DateTime CreatedAt { get; set; } = DateTime.Now;
        public DateTime? UpdatedAt { get; set; }

        [Required]
        public int Rating { get; set; } // Rating scale of 1-5

        [ForeignKey(nameof(Activity))]
        public int ActivityId { get; set; }
        public Activity Activity { get; set; } = null!;

        [ForeignKey(nameof(Member))]
        public int MemberId { get; set; }
        public Member Member { get; set; } = null!;
    }
}

File: ScoutTrack/ScoutTrack.UI/scouttrack_desktop/lib/models/review.dart
import 'package:json_annotation/json_annotation.dart';

part 'review.g.dart';

@JsonSerializable()
class Review {
  final int id;
  final String content;
  final DateTime createdAt;
  final DateTime? updatedAt;
  final int rating;
  final int activityId;
  final int memberId;
  final String activityTitle;
  final String memberName;

  Review({
    required this.id,
    required this.content,
    required this.createdAt,
    this.updatedAt,
    required this.rating,
    required this.activityId,
    required this.memberId,
    required this.activityTitle,
    required this.memberName,
  });

  factory Review.fromJson(Map<String, dynamic> json) => _$ReviewFromJson(json);
  Map<String, dynamic> toJson() => _$ReviewToJson(this);
}
